dashboard/app/models/user.rb

```diff
@@ -926,15 +948,47 @@ def self.send_reset_password_instructions(attributes={})
       return user
     end
 
-    user = find_by_email_or_hashed_email(attributes[:email]) || User.new(email: attributes[:email])
-    if user && user.persisted?
-      user.raw_token = user.send_reset_password_instructions(attributes[:email]) # protected in the superclass
-    else
-      user.errors.add :email, :not_found
+    email = attributes[:email]
+    associated_users = User.associated_users(email)
+    return User.new(email: email).send_reset_password_for_users(email, associated_users)
+  end
+
+  attr_accessor :child_users
+  def send_reset_password_for_users(email, users)
+    if users.empty?
+      not_found_user = User.new(email: email)
+      not_found_user.errors.add :email, :not_found
+      return not_found_user
+    end
+
+    # Normal case: single user, owner of the email attached to this account
+    if users.length == 1 && (users.first.email == email || users.first.hashed_email == User.hash_email(email))
+      primary_user = users.first
+      primary_user.raw_token = primary_user.send_reset_password_instructions(email) # protected in the superclass
+      return primary_user
+    end
+
+    # One or more users are associated with parent email, generate reset tokens for each one
```

Comment:
there's a bit of complicated/duplicated code here that could be refactored into a shared helper method (to keep simple and avoid going out of sync).